src/controllers/keystore/keystore.ts

```diff
@@ -696,7 +697,7 @@ export class KeystoreController extends EventEmitter implements IKeystoreControl
         }
       })
       // No need to re-add keys that are already added, private key never changes
-      .filter(({ addr, type }) => !keys.some((x) => x.addr === addr && x.type === type))
+      .filter(({ addr, type, meta }) => !keys.some((x) => x.addr === addr && x.type === type && x.meta.railgunType === meta.railgunType))
```

Comment:
my idea here is that we can fully reuse the keystore's `addKeys` method for 0zk keys as well, we just have to make this one small change in the logic (so that we don't count spending and viewing 0zk keys as duplicates and store both)